Description: Ignore warnings of GLPK 4.65 when parsing doctests
 GLPK 4.65 prints warnings which lead to failing doctests.
 Upstream does not use GLPK 4.65 yet. Ignore the warnings until a
 solution is found.
Author: Tobias Hansen <thansen@debian.org>
Bug: https://trac.sagemath.org/ticket/24824
Bug-Debian: https://bugs.debian.org/891465

--- a/sage/src/sage/doctest/parsing.py
+++ b/sage/src/sage/doctest/parsing.py
@@ -39,6 +39,7 @@
 
 float_regex = re.compile(r'\s*([+-]?\s*((\d*\.?\d+)|(\d+\.?))([eE][+-]?\d+)?)')
 optional_regex = re.compile(r'(py2|py3|long time|not implemented|not tested|known bug)|([^ a-z]\s*optional\s*[:-]*((\s|\w)*))')
+glpk_simplex_warning_regex = re.compile(r'(Long-step dual simplex will be used)')
 find_sage_prompt = re.compile(r"^(\s*)sage: ", re.M)
 find_sage_continuation = re.compile(r"^(\s*)\.\.\.\.:", re.M)
 find_python_continuation = re.compile(r"^(\s*)\.\.\.([^\.])", re.M)
@@ -1068,6 +1069,7 @@
             <type 'float'>
         """
         got = self.human_readable_escape_sequences(got)
+        got = glpk_simplex_warning_regex.sub('', got)
         if isinstance(want, MarkedOutput):
             if want.random:
                 return True
